frontend/src/Pages/Favorites.tsx

```diff
@@ -0,0 +1,150 @@
+                        channel_title?: string;
+                        provider_name?: string;
+                    }[];
+                }
+                const typedResponse = response as ApiResponse;
+                const mappedFavorites: CombinedFavorite[] =
+                    typedResponse.data.map((item) => ({
+                        id: item.id,
+                        content_id: item.content_id,
+                        name: item.name,
+                        type: item.type,
+                        thumbnail_url: item.thumbnail_url,
+                        description: item.description,
+                        visibility_status: item.visibility_status,
+                        open_content_provider_id: item.open_content_provider_id,
+                        content_url: item.content_url,
+                        channel_title: item.channel_title ?? '',
+                        provider_name: item.provider_name ?? ''
+                    }));
+                setFavorites(mappedFavorites);
+            } catch (err) {
+                setError('Error fetching favorites.');
+                console.error(err);
+            } finally {
+                setIsLoading(false);
+            }
+        };
+        void fetchFavorites();
+    }, [user]);
+
+    const handleUnfavorite = async (
+        contentId: number,
+        type: 'video' | 'library'
+    ) => {
+        try {
+            setIsLoading(true);
+            let endpoint = '';
+
+            if (type === 'video') {
+                endpoint = `videos/${contentId}/favorite`;
+            } else {
+                endpoint = `open-content/${contentId}/save`;
+            }
+
+            const favorite = favorites.find(
+                (fav) => fav.content_id === contentId && fav.type === type
+            );
+
+            if (!favorite) {
+                setError('Favorite not found.');
+                return;
+            }
+
+            const { id, name, open_content_provider_id } = favorite;
+            if (!name || !open_content_provider_id) {
+                setError('Favorite data is incomplete.');
+                return;
+            }
+
+            const payload = {
+                name,
+                content_id: contentId,
+                open_content_provider_id,
+                content_url: `/api/proxy/libraries/${contentId}/`
+            };
+
+            const response = await API.put(endpoint, payload);
+
+            if (response.success) {
+                await mutate('/open-content/favorites');
+                setFavorites((prevFavorites) =>
+                    prevFavorites.filter((fav) => fav.id !== id)
+                );
+
+                if (type === 'library') {
+                    await mutate('/api/libraries');
+                }
+            } else {
+                setError('Failed to unfavorite item.');
+            }
+        } catch (err) {
+            setError('Error unfavoriting item.');
```

Comment:
could we have a toast that sets this error? Im not sure if a success toast is necessary, but I am feeling like an error toast would be helpful for the user to know that they had an issue favoriting that item.